internal/statemachine/classic_states.go

```diff
@@ -70,20 +70,43 @@ func (stateMachine *StateMachine) parseImageDefinition() error {
 	}
 
 	// do custom validation for gadgetURL being required if gadget is not pre-built
-	if imageDefinition.Gadget.GadgetType != "prebuilt" && imageDefinition.Gadget.GadgetURL == "" {
-		jsonContext := gojsonschema.NewJsonContext("gadget_validation", nil)
-		errDetail := gojsonschema.ErrorDetails{
-			"key":   "gadget:type",
-			"value": imageDefinition.Gadget.GadgetType,
-		}
-		result.AddError(
-			imagedefinition.NewMissingURLError(
-				gojsonschema.NewJsonContext("missingURL", jsonContext),
-				52,
+	if imageDefinition.Gadget != nil {
+		if imageDefinition.Gadget.GadgetType != "prebuilt" && imageDefinition.Gadget.GadgetURL == "" {
+			jsonContext := gojsonschema.NewJsonContext("gadget_validation", nil)
+			errDetail := gojsonschema.ErrorDetails{
+				"key":   "gadget:type",
+				"value": imageDefinition.Gadget.GadgetType,
+			}
+			result.AddError(
+				imagedefinition.NewMissingURLError(
+					gojsonschema.NewJsonContext("missingURL", jsonContext),
+					52,
+					errDetail,
+				),
 				errDetail,
-			),
-			errDetail,
-		)
+			)
+		}
+	}
+
+	// don't allow any images to be created without a gadget
+	if imageDefinition.Gadget == nil {
+		if imageDefinition.Artifacts.Img == nil ||
+			imageDefinition.Artifacts.Iso == nil ||
+			imageDefinition.Artifacts.Qcow2 == nil {
+			jsonContext := gojsonschema.NewJsonContext("image_without_gadget", nil)
+			errDetail := gojsonschema.ErrorDetails{
+				"key1": "img:",
```

Comment:
This seems to hard-code `"img:"` right now, but I suppose this should be dependent on whichever artifact has been defined, right?